tensorflow_datasets/core/features/image_feature.py

```diff
@@ -186,34 +208,45 @@ def repr_html(self, ex: np.ndarray) -> str:
 
   def repr_html_batch(self, ex: np.ndarray) -> str:
     """`Sequence(Image())` are displayed as `<video>`."""
-    return make_video_repr_html(ex)
+    return make_video_repr_html(ex, self._use_colormap)
 
   @classmethod
   def from_json_content(cls, value: Json) -> 'Image':
     shape = tuple(value['shape'])
+    use_colormap = value['use_colormap']
     dtype = tf.dtypes.as_dtype(value['dtype'])
     encoding_format = value['encoding_format']
-    return cls(shape=shape, dtype=dtype, encoding_format=encoding_format)
+    return cls(
+        shape=shape,
+        dtype=dtype,
+        encoding_format=encoding_format,
+        use_colormap=use_colormap
+    )
 
   def to_json_content(self) -> Json:
     return {
         'shape': list(self._shape),
         'dtype': self._dtype.name,
-        'encoding_format': self._encoding_format
+        'encoding_format': self._encoding_format,
+        'use_colormap': self._use_colormap
     }
 
 
 # Visualization single image
 
 
-def _create_thumbnail(ex: np.ndarray) -> PilImage:
+def _create_thumbnail(ex: np.ndarray, use_colormap: bool) -> PilImage:
   """Creates the image from the np.array input."""
   PIL_Image = lazy_imports_lib.lazy_imports.PIL_Image  # pylint: disable=invalid-name
 
   _, _, c = ex.shape
   postprocess = _postprocess_noop
   if c == 1:
-    ex = ex.squeeze(axis=-1)
+    if use_colormap:
+      mode = None
+      image_with_colormap = _apply_colormap(ex)
+      ex = image_with_colormap * 255
```

Comment:
`ex = image_with_colormap * 255` should go inside `_apply_colormap` directly no ?